pkg/frontend/transport/handler.go

```diff
@@ -145,6 +145,8 @@ func (f *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 	if err != nil {
 		writeError(w, err)
+		queryString = f.parseRequestQueryString(r, buf)
+		f.reportFailedQueryStats(r, queryString, queryResponseTime, stats)
```

Comment:
can we include the error in the log message?

Reply:
I also ask if we shouldn't include an `err` field in the log message.